File: frontend/lib/models/asset.dart
class Asset {
  final int assetId;
  final int placeId;
  final String assetUrl;
  final String assetType;
  final int position;
  final DateTime createdAt;

  Asset({
    required this.assetId,
    required this.placeId,
    required this.assetUrl,
    required this.assetType,
    required this.position,
    required this.createdAt,
  });

  factory Asset.fromJson(Map<String, dynamic> json) => Asset(
    assetId: json['asset_id'],
    placeId: json['place_id'],
    assetUrl: json['asset_url'],
    assetType: json['asset_type'],
    position: json['position'],
    createdAt: DateTime.parse(json['created_at']),
  );
}

File: backend/internal/models/asset.go
package models

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type Asset struct {
	AssetID       uuid.UUID `gorm:"type:uuid;primaryKey;column:asset_id" json:"asset_id"`
	PlaceID       uuid.UUID `gorm:"type:uuid;not null;index;column:place_id" json:"place_id"`
	Place         Place     `gorm:"constraint:OnDelete:CASCADE;" json:"-"`
	AssetFilename string    `gorm:"column:asset_filename;not null" json:"asset_filename"`
	AssetType     string    `gorm:"not null;check:asset_type IN ('image', 'video')" json:"asset_type"`
	Position      int       `gorm:"default:0" json:"position"`
	CreatedAt     time.Time `gorm:"autoCreateTime" json:"created_at"`
}

func (a *Asset) BeforeCreate(tx *gorm.DB) error {
    if a.AssetID == uuid.Nil {
        a.AssetID = uuid.Must(uuid.NewV7())
    }
    return nil
}

func (Asset) TableName() string {
	return "assets"
}